src/components/common/LoadMoreButton/index.tsx

```diff
@@ -1,9 +1,45 @@
-import { type ReactElement } from 'react'
+import { type MutableRefObject, useCallback, useEffect, useRef, useState, type ReactElement } from 'react'
 import { Button } from '@mui/material'
 
-const LoadMoreButton = ({ onLoadMore }: { onLoadMore: () => void }): ReactElement => {
+const useIntersectionObserver = (element: MutableRefObject<HTMLElement | null>): boolean => {
+  const [isIntersecting, setIsIntersecting] = useState<boolean>(false)
+  const observer = useRef<IntersectionObserver | undefined>()
+
+  const callback = useCallback(([entry]: IntersectionObserverEntry[]) => {
+    setIsIntersecting(entry.isIntersecting)
+  }, [])
+
+  useEffect(() => {
+    observer.current?.disconnect()
+    observer.current = undefined
+
+    if (!element.current) return
+
+    observer.current = new IntersectionObserver(callback)
+
+    observer.current.observe(element.current)
+
+    return () => {
+      observer.current?.disconnect()
+      observer.current = undefined
+    }
+  }, [callback])
+
+  return isIntersecting
+}
+
+const LoadMoreButton = ({ onLoadMore, loading }: { onLoadMore: () => void; loading: boolean }): ReactElement => {
```

Comment:
I don't think we should render a button anymore, just skeletons.

Reply:
We could lower the `IntersectionObserver` [`threshold`](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API#creating_an_intersection_observer) to load sooner as well.